Validation/HGCalValidation/plugins/HGCalHitCalibration.cc

```diff
@@ -159,7 +159,14 @@ void HGCalHitCalibration::fillWithRecHits(
   if (hitmap.find(hitid) == hitmap.end()) {
     // Hit was not reconstructed
     IfLogTrace(debug_ > 0, "HGCalHitCalibration")
-      << ">>> Failed to find detid " << hitid.rawId()
+      << ">>> Failed to find detid " << std::hex << hitid.rawId() << std::dec
+      << " Det " << hitid.det()
+      << " Subdet " << hitid.subdetId() << std::endl;
+    return;
+  }
+  if (hitid.det() != DetId::Forward) {
```

Comment:
this addition is apparently responsible for the DQM changes noticed earlier
https://github.com/cms-sw/cmssw/pull/21440#issuecomment-351858391
this change looks somewhat straightforward.

Reply:
The following code works for the silicon part only and needs to be protected with this.

Reply:
I see that there are more entries in the calib histograms now
e.g. in PU200 test (wf 20234.2)
<img width="826" alt="wf20234p2pu200_hgcalcalib_eop_100" src="https://user-images.githubusercontent.com/4676718/34421009-a5456ed8-ebc1-11e7-851b-eaf6c31e52d7.png">

this apparently corresponds to situations where the max energy rechit (from a set of detids matching to sim truth detIds for a given calo particle) is in the "Hcal" part of the detector. Depending on the underlying logic, such sim calo particles may not be desired and should not show up as if they had an energy peaking in the silicon detectors.

This is a validation choice selection issue though, not directly connected to the reco signature.